src/test/scala/com/advancedtelematic/campaigner/util/CampaignerSpec.scala

```diff
@@ -2,14 +2,15 @@ package com.advancedtelematic.campaigner.util
 
 import com.advancedtelematic.libats.test.LongTest
 import org.scalacheck.Gen
+import org.scalacheck.rng.Seed
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.{FlatSpecLike, Matchers}
 
 trait CampaignerSpecUtil {
   implicit class GenerateOps[T](value: Gen[T]) {
-    def generate: T = value.sample.get
+    def generate: T = value.pureApply(Gen.Parameters.default, Seed.random(), retries = 100)
```

Comment:
In device-registry we also have such a class:
```
object GeneratorOps {
  implicit class GenSample[T](gen: Gen[T]) {
    @tailrec
    final def generate: T =
      gen.sample match {
        case Some(v) => v
        case None    => generate
      }
  }
}
```
I don't know if there is any big advantage of one over the other, but the one from dev-reg is somewhat easier to read?
I guess at some point we can move this to libats.

Reply:
Yeah, while it might be a bit easier to read, it has one disadvantage: it won't stop until it generates something. It shouldn't be a problem most of the time, but there is still a chance that it won't stop ever. `pureApply` on the other hand has a fixed limit of retries.